### PR TITLE
AED greedy confidence estimation

### PR DESCRIPTION
**Pull request by GNroy**
# What does this PR do ?

Adds greedy confidence estimation for AED multitask models

**Collection**: [ASR]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/modules/transformer/transformer_generators.py [hunk 1]
```diff
@@ -198,16 +241,29 @@ def _forward(
             pad_profile = torch.max(pad_profile, (next_tokens == self.eos).long())
             tgt = torch.cat((tgt, next_tokens.unsqueeze(1)), dim=-1)
 
+            if self.preserve_step_confidence:
+                if i == 0:
+                    step_confidence[0] = step_confidence[0].to(dtype=logits.dtype)
```
Comment (pzelasko) [inline nemo/collections/asr/modules/transformer/transformer_generators.py:246]: Set the right dtype at initialization (encoder_hidden_states or decoder_input_ids should already have the right dtype)

**Review by pzelasko — APPROVED** [review]
LGTM, left one minor comment

File: nemo/collections/asr/modules/transformer/transformer_generators.py (hunk 1 above)
Comment (GNroy) [reply nemo/collections/asr/modules/transformer/transformer_generators.py]: @pzelasko thanks, fixed!

**Review by pzelasko — APPROVED** [review]
